pkg/autoscaler/metrics/collector.go

```diff
@@ -285,50 +292,56 @@ func newCollection(metric *av1alpha1.Metric, scraper StatsScraper, tickFactory f
 			metric.Spec.PanicWindow, config.BucketSize),
 		scraper: scraper,
 
-		stopCh: make(chan struct{}),
+		tickFactory: tickFactory,
 	}
 
-	key := types.NamespacedName{Namespace: metric.Namespace, Name: metric.Name}
-	logger = logger.Named("collector").With(zap.String(logkey.Key, key.String()))
+	if metric.Spec.ScrapeTarget != "" {
```

Comment:
Aren't metrics being updated every reconcile loop? Would there be a case where the user changes the TBC to non -1?

Reply:
Theoretically (changing the PA), yes. Practically, this involved changing the revision and thus stamping out a new PA.

This piece of code wouldn't be called by the metric reconcilation though. It'd be the "updateMetric" function, which would enable the scraper or not.